python/ray/experimental/channel/torch_tensor_type.py

```diff
@@ -40,6 +42,10 @@ def __init__(
                 host memory, using numpy as the serialization format. Pass
                 TorchTensorType.NCCL or "nccl" to use NCCL instead, avoiding
                 the host memory copy.
+            device: Defines the target device for transporting a torch.Tensor.
+                If "default" is set, it retains the same device type as in the sender.
+                If "cpu" is set, it moves the tensor to the CPU on the receiver.
+                If "gpu" or "cuda" is set, it moves the tensor to the GPU on the receiver.
```

Comment:
Let's add a note here that transport="nccl" will error if the device here is "cpu".